src/dosext/mfs/mfs.c

```diff
@@ -1335,12 +1336,17 @@ static struct dir_list *get_dir(char *name, char *mname, char *mext, int drive)
   char fname[8];
   char fext[3];
 
-  if(is_dos_device(name) || !find_file(name, &sbuf, drive, NULL))
+  if (is_dos_device(name) || !find_file(name, &sbuf, drive, NULL)) {
+    if (!dir_list)
```

Comment:
You meant to say `if (patherror)`?

Reply:
I did, and just fixed it. Too much of a rush!